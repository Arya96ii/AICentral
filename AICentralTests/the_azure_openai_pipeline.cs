using System.Net;
using System.Text;
using AICentralTests.TestHelpers;
using ApprovalTests;
using Newtonsoft.Json;
using Shouldly;
using Xunit.Abstractions;

namespace AICentralTests;

public class the_azure_openai_pipeline : IClassFixture<TestWebApplicationFactory<Program>>

{
    private readonly TestWebApplicationFactory<Program> _factory;
    private readonly HttpClient _httpClient;

    public the_azure_openai_pipeline(TestWebApplicationFactory<Program> factory, ITestOutputHelper testOutputHelper)
    {
        _factory = factory;
        factory.OutputHelper = testOutputHelper;
        _httpClient = factory.CreateClient();
    }

    [Fact]
    public async Task handles_images()
    {
        var result = await _httpClient.PostAsync(
            "/openai/deployments/api-key-auth-fake-image/images/generations?api-version=2023-05-15",
            new StringContent(JsonConvert.SerializeObject(new
            {
            }), Encoding.UTF8, "application/json"));

        result.StatusCode.ShouldBe(HttpStatusCode.OK);
        var content = await result.Content.ReadAsStringAsync();
        Approvals.VerifyJson(content);
    }
}