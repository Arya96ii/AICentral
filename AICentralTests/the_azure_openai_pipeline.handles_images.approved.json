{
    "Created": 1699326600,
    "Data": [
        {
            "url": "https://fakeimage.localtest.me/some-image"
        }
    ]
}